front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx

```diff
@@ -165,144 +73,200 @@ const RollingStockEditorForm = ({
+          >
+            <InputGroupSNCF
+              id={property.title}
+              label={t(`${property.title}`)}
+              typeValue={property.type}
+              type={optionValue}
+              value={rollingStockValues[property.title] as string | number}
+              options={property.units.map((unit) => ({
+                id: `${property.title}-${unit}`,
+                label: unit,
+              }))}
+              handleType={(type) => {
+                setRollingStockValues({
+                  ...rollingStockValues,
+                  [property.title]: type.value,
+                } as SetStateAction<RollingStockParametersValues>);
+                setOptionValue(type.type as string);
+              }}
+              min={property.min}
+              max={property.max}
+              isInvalid={
+                rollingStockValues[property.title]?.toString().length === 0 ||
+                (rollingStockValues[property.title] as number) < (property.min as number) ||
+                (rollingStockValues[property.title] as number) > (property.max as number)
+              }
+              errorMsg={
+                property.max
+                  ? t('errorMessages.minMaxRangeError', { min: property.min, max: property.max })
+                  : t('errorMessages.minRangeError', { min: property.min })
+              }
+              step="any"
+              sm
+              condensed
+              orientation="right"
             />
-          )}
-        </div>
-      ));
+          </div>
+        ) : (
+          <InputSNCF
+            id={property.title}
+            name={property.title}
+            label={t(`${property.title}`)}
+            type={property.type}
+            step="any"
+            min={property.min}
+            max={property.max}
+            isInvalid={
+              Number.isNaN(rollingStockValues[property.title]) ||
+              (rollingStockValues[property.title] as number) < (property.min as number) ||
+              (rollingStockValues[property.title] as number) > (property.max as number)
+            }
+            errorMsg={
+              property.max
+                ? t('errorMessages.minMaxRangeError', { min: property.min, max: property.max })
+                : t('errorMessages.minRangeError', { min: property.min })
+            }
+            unit={property.unit}
+            value={rollingStockValues[property.title] as number}
+            onChange={(e) =>
+              setRollingStockValues({
+                ...rollingStockValues,
+                [property.title]: parseFloat(e.target.value),
+              })
+            }
+            sm
+            isFlex
+            key={property.title}
+          />
+        );
+      });
 
-  const queryArg = (data: FieldValues): RollingStockUpsertPayload => ({
-    railjson_version: data.railjson_version,
+  const queryArg = (data: RollingStockParametersValues): RollingStockUpsertPayload => ({
```

Comment:
Move this to the utils file, to decrease the number of lines :)

Reply:
done